ciri/api/api.c

```diff
@@ -144,9 +95,39 @@ retcode_t iota_api_find_transactions(find_transactions_req_t const *const req,
   return RC_OK;
 }
 
-retcode_t iota_api_get_trytes(get_trytes_req_t const *const req,
+retcode_t iota_api_get_trytes(iota_api_t const *const api,
+                              get_trytes_req_t const *const req,
                               get_trytes_res_t *const res) {
-  return RC_OK;
+  retcode_t ret = RC_OK;
+  flex_hash_array_t *iter = NULL;
+  flex_trit_t tx_trits[FLEX_TRIT_SIZE_8019];
+  tryte_t tx_trytes[NUM_TRYTES_SERIALIZED_TRANSACTION + 1];
+  DECLARE_PACK_SINGLE_TX(tx, txp, pack);
+  size_t nbr = 0;
+
+  LL_FOREACH(req->hashes, iter) {
+    hash_pack_reset(&pack);
+    if ((ret = iota_tangle_transaction_load(api->tangle, TRANSACTION_FIELD_HASH,
+                                            iter->hash, &pack)) != RC_OK) {
+      return ret;
+    }
+    if (pack.num_loaded != 0) {
+      transaction_serialize_on_flex_trits(txp, tx_trits);
+      flex_trits_to_trytes(tx_trytes, NUM_TRYTES_SERIALIZED_TRANSACTION,
+                           tx_trits, NUM_TRITS_SERIALIZED_TRANSACTION,
+                           NUM_TRITS_SERIALIZED_TRANSACTION);
+    } else {
+      memset(tx_trytes, '9', NUM_TRYTES_SERIALIZED_TRANSACTION);
+    }
+    // TODO Remove when cclient uses hash containers
+    tx_trytes[NUM_TRYTES_SERIALIZED_TRANSACTION] = '\0';
+    get_trytes_res_add_trytes(res, tx_trytes);
+    nbr++;
+  }
+  if (nbr > api->max_get_trytes) {
```

Comment:
this check should happen inside or before the loop?

Reply:
Absolutely